cmd/agent/main.go

```
@@ -286,6 +289,16 @@ func startLogger(workers int, logger *zap.SugaredLogger) *loggerArgs {
 		os.Exit(-1)
 	}
 
+	var annotationKVPair map[string]string = map[string]string{}
+	for _, annotations := range *metadataAnnotations {
+		k, v, found := strings.Cut(annotations, "=")
+		if found {
+			annotationKVPair[k] = v
+		} else {
+			logger.Warnf("annotation does not adhere to desired format got key: %s value: %s", k, v)
```

I think it is better to error out and exit the agent if the annotation is not specified correctly instead of silently printing out the warnings.

---

done; agent now exits when a malformed annotation pair is found